### PR TITLE
Newapp: Allow Docker FROM in Dockerfile to point to an image stream or invalid image

### PR DESCRIPTION
**Pull request by csrwng**
- Allows a match to an image stream by name without a matching tag. 
- Allows matching the Dockerfile FROM image to an image stream tag
- Allows Dockerfiles with FROM tags that don't match any images and prints a warning.

### CODE REVIEW

**Comment by csrwng** [comment]
@ncdc @bparees ptal
@kwoodson fyi

File: pkg/generate/app/imagelookup.go [hunk 1]
```diff
@@ -299,6 +300,16 @@ func (r ImageStreamResolver) Resolve(value string) (*ComponentMatch, error) {
 		if latest == nil {
 			// continue searching in the next namespace
 			glog.V(2).Infof("no image recorded for %s/%s:%s", repo.Namespace, repo.Name, searchTag)
+			matches = append(matches, &ComponentMatch{
+				Value:       ref.String(),
+				Argument:    fmt.Sprintf("--image=%q", ref.String()),
+				Name:        ref.Name,
+				Description: fmt.Sprintf("Image repository %s (tag %q) in project %s, tracks %q", repo.Name, searchTag, repo.Namespace, repo.Status.DockerImageRepository),
```
Comment (bparees) [inline pkg/generate/app/imagelookup.go]: image stream?

File: pkg/generate/app/imagelookup.go [hunk 2]
```diff
@@ -299,6 +300,16 @@ func (r ImageStreamResolver) Resolve(value string) (*ComponentMatch, error) {
...
+				Score:       0.5,
+
+				ImageStream: repo,
```
Comment (bparees) [inline pkg/generate/app/imagelookup.go]: stream

File: pkg/generate/app/imagelookup.go [hunk 3]
```diff
@@ -299,6 +300,16 @@ func (r ImageStreamResolver) Resolve(value string) (*ComponentMatch, error) {
 		if latest == nil {
 			// continue searching in the next namespace
 			glog.V(2).Infof("no image recorded for %s/%s:%s", repo.Namespace, repo.Name, searchTag)
+			matches = append(matches, &ComponentMatch{
+				Value:       ref.String(),
+				Argument:    fmt.Sprintf("--image=%q", ref.String()),
```
Comment (bparees) [inline pkg/generate/app/imagelookup.go]: can we make a new arg --image-stream and deprecate this one?

**Comment by bparees** [comment]
couple nits, otherwise lgtm.

**Comment by csrwng** [comment]
comments addressed

**Comment by bparees** [comment]
lgtm.

**Comment by csrwng** [comment]
[merge]

**Pull request merged**